src/common/TokenParser.ts

```diff
@@ -151,20 +152,33 @@ export class TokenParser {
                 {from: address}
             ]
         }).select("contract").exec().then((operations: any) => {
-            const tokenContracts = this.extractTokenContracts(operations);
-            const contractDetailsPromises = this.getContractDetails(tokenContracts);
-            return Promise.all(contractDetailsPromises).then((tokens: any[]) => {
-                const balancePromises: any = [];
-
-                // get the balances
-                tokens.map((token: any) => {
-                    balancePromises.push(getTokenBalanceForAddress(address, token.address, token.decimals));
+            const tokenContracts: any[] = this.extractTokenContracts(operations);
+            const contractDetailsPromises: any[] = this.getContractDetails(tokenContracts);
+
+            return Promise.map(contractDetailsPromises, (contract) => {
+                return this.getTokenBalance(address, contract.address).then((balance: string) => {
+                    return {
+                        name: contract.name,
+                        balance,
+                    }
                 });
-                return Promise.all(balancePromises);
             });
         });
     }
 
+    private getTokenBalance(address: string, contractAddress: string) {
+        const tokenAddress: string = address.substring(2);
+        const contractData: string = `0x70a08231000000000000000000000000${tokenAddress}`;
```

Comment:
What is this `contractData`?

Reply:
Combination of contract 'balanceOf()' ERC20 token function in hex and tokenAddress, required to call the balance of an address.

Reply:
Fixed in next PR